scripts/map_genbank.sh

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+set -beEu -o pipefail
+
+metadata_csv=$1
+fasta_dir=$2
+
+# Run in a temporary directory, remove when done
+tmpdir=$(mktemp --directory --tmpdir map_genbank.XXXXXXXX)
+cd $tmpdir
+
+# Use NCBI Virus search to download metadata for H5N1 collected in USA since 2023
+# Influenza A virus NCBI Taxonomy ID = 11320
+tax_id=11320
+ncbi_virus_url='https://www.ncbi.nlm.nih.gov/genomes/VirusVariation/vvsearch2/?fq=%7B%21tag%3DSeqType_s%7DSeqType_s%3A%28%22Nucleotide%22%29&fq=VirusLineageId_ss%3A%28'$tax_id'%29&fq=%7B%21tag=CollectionDate_dr%7DCollectionDate_dr:%5B2023-01-01T00:00:00.00Z%20TO%202024-06-04T23:59:59.00Z%5D&fq=%7B%21edismax%20qf=Serotype_s%7DH5N1&fq=%7B%21tag=Country_s%7DCountry_s:%28%22USA%22%29&q=%2A%3A%2A&cmd=download&dlfmt=csv&fl=genbank_acc%3AAccVer_s%2Cisolate%3AIsolate_s%2Clocation%3ACountryFull_s%2Ccollected%3ACollectionDate_s%2Chost%3AHost_s%2Cbioproject_accession%3ABioProject_s%2Cstrain%3AStrain_s%2Cserotype%3ASerotype_s%2Csegment%3ASegment_s&sort=id+asc'
```

Comment:
If I'm reading this correctly, a collection date range seems to be specified in the URL. Was this added on purpose?
```
CollectionDate_dr%7DCollectionDate_dr:%5B2023-01-01T00:00:00.00Z%20TO%202024-06-04T23:59:59.00
```

Reply:
Yes, in order to limit the volume of data returned by the query, I added a constraint that the sample had to be collected since 2023 (this is from using https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?SeqType_s=Nucleotide&VirusLineage_ss=Influenza%20A%20virus,%20taxid:11320 and observing the network requests)... oh, but I see it is a bug to keep the end date from that day of 2024-06-04!

Thanks for catching that.  I will see if there is a way to specify no end date in the query, and if not, to use the current date.

Reply:
I changed the script to use today's date as the end of the range.  Since this script is very specific to the USDA sample naming scheme, it would probably also work to use the BioProject accession in the query instead of collection date.  Thanks again for catching that and let me know if you would like any improvements.

Reply:
Thank you for making the change. We can stick with using the date as it doesn't really matter which one we use.